.github/workflows/main.yml

```diff
@@ -18,7 +18,9 @@ jobs:
         - { name: Windows (mingw32),      os: windows-latest, shell: 'msys2 {0}', msystem: mingw32, msys-env: mingw-w64-i686 }
         - { name: Windows (mingw64),      os: windows-latest, shell: 'msys2 {0}', msystem: mingw64, msys-env: mingw-w64-x86_64 }
         - { name: Linux,                  os: ubuntu-20.04,   shell: sh,   flags: -GNinja }
-        - { name: MacOS,                  os: macos-latest,   shell: sh }
+        - { name: MacOS (x64),            os: macos-latest,   shell: sh,   flags: -DCMAKE_OSX_ARCHITECTURES=x86_64 }
```

Comment:
Is there any benefit to having these here even with the Universal build?

Reply:
That's a fair point. Personally, I would argue that keeping standalone workflows for x64 and arm64 would make more apparent issues such as:

- Did compilation fail only on one architecture, if so which?
- Did compilation fail because I explicitly asked for a universal build?

When running CMake with both `OSX_ARCHITECTURES` set, it takes the approach of calling Apple Clang with both `-arch arm64` and `-arch x86_64` instead of building it twice and merging the result with lipo. 

So if, for example, you get a compilation issue only on arm64, the universal binary compilation will be considered as failed, but your error message will most likely not indicate the architecture in which the issue happened.

<details>
<summary> Here's an example code that only fails on arm64 and its associated issue </summary>

```cpp
#include <stdio.h>

#ifdef __x86_64__
#define GREETING "Hello!"
#elif __aarch64__
#define GREETING 42
#endif

void greet(void){
    puts(GREETING);
}
```

```
error: no matching function for call to 'puts'
    puts(GREETING);
    ^~~~
```

</details>

Reply:
> When running CMake with both OSX_ARCHITECTURES set, it takes the approach of calling Apple Clang with both -arch arm64 and -arch x86_64 instead of building it twice and merging the result with lipo.

I'm surprised this is valid / allowed for complex binaries like SDL - it's not the workflow Apple recommends at least, as far as I know.

Reply:
I'm surprised this works too.

Reply:
This may be a leftover of the previous universal binary format with PowerPC + i386 which CMake also supported back in the days. Back then, It seemed to be either "Tick a box in Xcode" or ["add both `-arch` flags"](https://web.archive.org/web/20080513183551if_/http://developer.apple.com/documentation/MacOSX/Conceptual/universal_binary/universal_binary_compiling/chapter_2_section_7.html#//apple_ref/doc/uid/TP40002217-CH206-BAJJFEAC). Digging through CMake's source, I saw that `lipo -create` is only called for iOS builds.

The only CMake generator that will build arm64 and x86_64 independently is Xcode. Unix Makefiles and Ninja will add both `-arch` to your CFLAGS.